dragonfly/block/instrument/instrument.go

```diff
@@ -0,0 +1,93 @@
+package instrument
+
+// Instrument represents a note block instrument.
+type Instrument struct {
+	instrument
+}
+
+type instrument int32
+
+// MagicNumber ...
+func (i instrument) MagicNumber() int32 {
```

Comment:
I'm not so much a fan of `MagicNumber`. Might be better to just call it `Int32()`.